src/operator/pooling-inl.h

```diff
@@ -179,22 +184,37 @@ class PoolingProp : public OperatorProperty {
                   std::vector<TShape> *aux_shape) const override {
     CHECK_EQ(in_shape->size(), 1);
     const TShape &dshape = (*in_shape)[0];
-    CHECK_EQ(dshape.ndim(), 4) << \
-                               "Pooling: Input data should be 4D in (batch, channel, y, x)";
+    CHECK_GE(dshape.ndim(), 4) << "Pooling: Input data should be 4D in (batch, channel, y, x) "
+                               << "Or 5D in (batch, channel, d, y, x)";
     TShape oshape = dshape;
     if (dshape.ndim() ==  0) return false;
-    if (param_.global_pool) {
-      oshape[2] = 1;
-      oshape[3] = 1;
-    } else {
-      oshape[2] = std::min(dshape[2] + 2 * param_.pad[0] - param_.kernel[0] + param_.stride[0] - 1,
-                          dshape[2] + 2 * param_.pad[0] - 1) / param_.stride[0] + 1;
-      oshape[3] = std::min(dshape[3] + 2 * param_.pad[1] - param_.kernel[1] + param_.stride[1] - 1,
-                          dshape[3] + 2 * param_.pad[1] - 1) / param_.stride[1] + 1;
+    if (param_.kernel.ndim() == 2) {
+      CHECK_EQ(dshape.ndim(), 4) << "Pooling: Input data should be 4D in (batch, channel, y, x)";
+      if (param_.global_pool) {
+        oshape[2] = 1;
+        oshape[3] = 1;
+      } else {
+        oshape[2] = 1 + (dshape[2] + 2 * param_.pad[0] - param_.kernel[0]) / param_.stride[0];
```

Comment:
Hi, I got some problem when testing old code with new pooling interface as the datatype used here is unsigned int.  Maybe we need some more fool proof stuff here. This issue can be fixed if I claim it as global_pool.

* given input shape of 6 and kernel size 7. it can exit with alert of "kernel size exceed input".
* given input shape of 6 and kernel size 8, you can sneak through this check with 4294967296(max value of unsigned int).